rgw/dedup: add throttling mechanism (#64730)
ceph/ceph: doc/radosgw/s3_objects_dedup.rst

virtually always stored on a fast medium (SSD with heavy memory
caching).

The admin can throttle the estimate process by setting a limit to the number of
bucket-index reads per-second (each read brings 1000 object entries) using:

> $ radosgw-admin dedup throttle --max-bucket-index_ops=<count>

Reviewer: ```suggestion
$ radosgw-admin dedup throttle --max-bucket-index-ops=<count>
```

> A typical system performs about 100 bucket-index reads per second (i.e. 100,000 object entries).

Reviewer: ```suggestion
For example, if your system performs about 100 bucket-index reads per second (i.e. 100,000 object entries).
```

Author: actually, I think the number 100 is a good rule of thumb for the number of bucket-index reads per-second and not only an example

Setting the count to 50 will typically slow down access by half and so on...

> 

Reviewer: also, would be great if you explained about the data ops throttling - even if you are using an existing conf options for that.
you can reference them in the doc using `confval`. e.g.
```
.. confval::rgw_max_copy_obj_concurrent_io
```
